tools/autograd/load_derivatives.py

```diff
@@ -144,15 +144,25 @@ def set_up_derivatives(defn_name, defn, declaration):
 
         # Set up the derivative information
         derivatives = []
+        args_with_no_gradients = []
         for raw_names in sorted(defn.keys()):
             formula = defn[raw_names]
             names = split_names(raw_names)
-            derivatives.append(create_derivative(declaration, formula, names))
+            derivative = create_derivative(declaration['arguments'], declaration['returns'],
+                                           declaration['name'], formula, names)
+            if formula.lower().strip() == 'not_differentiable':
+                assert not sum([type(var_name) == list
+                                for var_name in derivative['var_names']]), \
+                    "Variable names associated to a formula should be a flat list"
+                args_with_no_gradients += derivative['var_names']
+            else:
+                derivatives.append(derivative)
+        args_with_gradients = list(filter(lambda x: x['name'] not in args_with_no_gradients, args_with_gradients))
```

Comment:
I still don't know what this is supposed to be doing -- it seems like if something is in both args_with_gradients and args_with_no_gradients it should be error, not that it should silently removed from args_with_gradients.